### PR TITLE
REF: implement c_FreqGroup, FreqGroup in tslibs.dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
These are already used in libperiod, better to be explicit about it.

Coming up: make FreqGroup into an enum with helper methods to convert between FreqGroup/PeriodDtype/DateOffset/Resolution

### CODE REVIEW

File: pandas/_libs/tslibs/dtypes.pyx [hunk 1]
```diff
@@ -106,3 +106,20 @@ _period_code_map.update({
     "W": 4000,   # Weekly
     "C": 5000,   # Custom Business Day
 })
+
+
+class FreqGroup:
```
Comment (WillAyd) [inline pandas/_libs/tslibs/dtypes.pyx:111]: Maybe overlooking something simple but do we need both or does the enum alone suffice?

**Review by WillAyd — COMMENTED** [review]
Question but on board with the concept

File: pandas/_libs/tslibs/dtypes.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/dtypes.pyx:111]: the one in the .pxd we can't access from python-space

Comment (jbrockmendel) [reply pandas/_libs/tslibs/dtypes.pyx:111]: also i expect before long we'll make this one into `class FreqGroup(Enum)` to de-duplicate it with Resolution

**Review by WillAyd — APPROVED** [review]

**Pull request merged**